tests/devapps/MauiApps/MauiAppBasic/MainPage.xaml.cs

```diff
@@ -15,30 +15,66 @@ public MainPage()
 
         private async void OnSignInClicked(object sender, EventArgs e)
         {
-            AuthenticationResult result = null;
-
             try
             {
                 PCAWrapper.Instance.UseEmbedded = this.useEmbedded.IsChecked;
                 // attempt silent login.
-                // If this is very first time and the device is not enrolled, it will throw MsalUiRequiredException
-                // If the device is enrolled, this will succeed.
-                result = await PCAWrapper.Instance.AcquireTokenSilentAsync(PCAWrapper.Scopes).ConfigureAwait(false);
+                // If this is very first time or user has signed out, it will throw MsalUiRequiredException
+                AuthenticationResult result = await PCAWrapper.Instance.AcquireTokenSilentAsync(PCAWrapper.Scopes).ConfigureAwait(false);
+
+                // call Web API to get the data
+                string data = await CallWebAPIWithToken(result).ConfigureAwait(false);
 
-                await ShowMessage("First AcquireTokenTokenSilent call", result.AccessToken).ConfigureAwait(false);
+                // show the data
+                await ShowMessage("AcquireTokenTokenSilent call", data).ConfigureAwait(false);
             }
             catch (MsalUiRequiredException)
             {
-                // This executes UI interaction ot obtain token
-                result = await PCAWrapper.Instance.AcquireTokenInteractiveAsync(PCAWrapper.Scopes).ConfigureAwait(false);
+                // This executes UI interaction to obtain token
+                AuthenticationResult result = await PCAWrapper.Instance.AcquireTokenInteractiveAsync(PCAWrapper.Scopes).ConfigureAwait(false);

```

Comment:
What happens when user cancels the interaction?

Reply:
It will throw an exception.